tests/test_reference_data.py

```diff
@@ -1,90 +1,73 @@
 
-    @patch.object(reference_data.Downloader, 'download_file')
-    def test_download_data(self, mocked_download):
-        files = [
-            'release-1/fasta/a_species/A_species.a_genome.dna.toplevel.fa.gz',
-            'release-1/variation/vcf/A_species.vcf.gz',
-            'release-1/variation/vcf/A_species.vcf.gz.tbi'
-        ]
-        self.downloader.ftp.nlst = Mock(
-            side_effect=[
-                ['dna'],
-                files[0:1],
-                files[1:]
-            ]
-        )
-        self.downloader.__dict__['ensembl_release'] = 'release-1'
-        self.downloader.download_data()
-        for f in files:
-            mocked_download.assert_any_call(f)
+        self.downloader.__dict__['_logger'] = Mock()
 
     @patch('builtins.input', return_value=None)
     @patch('os.path.isfile', side_effect=[False, True, False])
     @patch('subprocess.check_call')
     @patch('bin.reference_data.Downloader.run_background')
-    @patch('egcg_core.util.find_file')
+    @patch('egcg_core.util.find_file', return_value=None)
     @patch('egcg_core.util.find_files', return_value=[])
     def test_prepare_data(self, m_find_files, m_find_file, m_run, m_check_call, m_isfile, m_input):
-        m_find_file.side_effect = [
-            'a_genome.dna.toplevel.fa.gz',
-            'a_genome.dna.toplevel.fa',
-            None,  # no fa.fai, so run faidx
-            None,  # no .dict, so run picard
-            None,  # no .bwt, so run bwa index
+        self.downloader.reference_fasta = 'a_genome.dna.toplevel.fa'
+        self.downloader.reference_variation = 'a_species.vcf.gz'
+        m_input.side_effect = [
             'a_species.vcf.gz'
         ]
 
         self.downloader.prepare_data()
 
-        m_check_call.assert_any_call(['gzip', '-d', 'a_genome.dna.toplevel.fa.gz']),
         m_check_call.assert_any_call('gzip -dc a_species.vcf.gz > tmp.vcf', shell=True),
         m_check_call.assert_any_call('path/to/bgzip -c tmp.vcf > a_species.vcf.gz', shell=True)
-        assert m_check_call.call_count == 3
+        assert m_check_call.call_count == 2
 
         m_run.assert_any_call('path/to/samtools faidx a_genome.dna.toplevel.fa', 'faidx.log'),
-        m_run.assert_any_call('path/to/picard CreateSequenceDictionary R=a_genome.dna.toplevel.fa O=a_genome.dna.toplevel.dict', 'create_sequence_dict.log'),
+        m_run.assert_any_call(
+            'path/to/picard -Xmx20G CreateSequenceDictionary R=a_genome.dna.toplevel.fa O=a_genome.dna.toplevel.dict',
+            'create_sequence_dict.log'),
         m_run.assert_any_call('path/to/bwa index a_genome.dna.toplevel.fa', 'bwa_index.log'),
         m_run.assert_any_call('path/to/tabix -p vcf a_species.vcf.gz', 'tabix.log'),
-        m_run.assert_any_call('java -jar path/to/gatk -T ValidateVariants -V a_species.vcf.gz -R a_genome.dna.toplevel.fa -warnOnErrors', 'a_species.vcf.gz.validate_variants.log')
-        assert m_run.call_count == 5
+        assert m_run.call_count == 4
```

Comment:
There are some expected call_counts that have been reduced by 1 here. I assume this is because the `gzip -d` and `ValidateVariants` have moved elsewhere?

Reply:
Yes that is correct `gzip -d` has moved to ``prepare_data` and `ValidateVariants` has moved to `validate_data`.
I've add some docstrings to clarify.